dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/DotFolderTransformerBuilder.java

```diff
@@ -0,0 +1,103 @@
+package com.dotmarketing.portlets.contentlet.transform;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Builder to gain access to the Folder Transformer
+ */
+public class DotFolderTransformerBuilder {
+
+    private FolderAPI folderAPI = APILocator.getFolderAPI();
+
+    private User user;
+    private Role[] roles;
+    private List<String> folderIds;
+    private List<Folder> folders;
+
+    /**
+     * Folders can be specified with ids or folder objects
+     * @param folders
+     * @return
+     */
+    public DotFolderTransformerBuilder withFolders(final Folder... folders){
+       this.folders = Arrays.asList(folders);
+       return this;
+    }
+
+    /**
+     * Folders can be specified with ids or folder objects
+     * @param folders
+     * @return
+     */
+    public DotFolderTransformerBuilder withFolders(final List<Folder> folders){
+        this.folders = folders;
+        return this;
+    }
+
+    /**
+     * Folders can be specified with ids or folder objects
+     * @param folderIds
+     * @return
+     */
+    public DotFolderTransformerBuilder withFolders(final String... folderIds){
+        this.folderIds = Arrays.asList(folderIds);
+        return this;
+    }
+
+    /**
+     * Submit user and roles required to build the SiteBrowse view
+     * @param user
+     * @param roles
+     * @return
+     */
+    public DotFolderTransformerBuilder withUserAndRoles(final User user, final Role... roles){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The user-supplied array 'roles' is stored directly.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)